### PR TITLE
Create pull_request_template.md

### LINKED ISSUE #170: create pull request template
When a PR is created, this should automatically populate as the PR description
comment. If we're trying to adhere to this form, it's probably good to be
reminded of it.

### PR DESCRIPTION
**Pull request by barter-simsum**
### Description

Resolves #170 

When a PR is created, this should automatically populate as the PR description
comment. If we're trying to adhere to this form, it's probably good to be
reminded of it.

Note, this doesn't take effect unless added to /master.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+Review https://github.com/urbit/vere/blob/develop/CONTRIBUTING.md
+
+-->
+
+### Description
+
+Resolves # <!-- All PRs must link an issue. Do it here. -->
+
+A thorough description of the changes made.
```
Comment (zalberico) [inline .github/pull_request_template.md:11]: We probably want this to be a comment too.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+### Other sections
+
+...
+
+### Related
+
+References to any related issues, links, papers, etc.
```
Comment (zalberico) [inline .github/pull_request_template.md:19]: Same with this one (comment)

**Review by zalberico — CHANGES_REQUESTED** [review]
We probably want the two lines from the docs to also be comments, otherwise seems fine to me. Thanks for thinking of this.

**Comment by barter-simsum** [comment]
@Fang- , opposed?

**Comment by Fang-** [comment]
Yes, somewhat opposed. I've voiced my reservations about gh templates in the past. Especially for PRs (as opposed to issues) I think it's fair to expect a higher standard of description without hand-holding the author into it. To me, pre-filled templates always feel oppressive rather than helpful.

Just a gentle nudge to "please review contributing.md" as is at the top here is probably the least invasive thing we could do. As long as that's there, not sure what purpose it serves to also pre-write the section headings. The "Other sections" heading feels doubly out of place, since it's not even in the contributing.md.

Also still not a fan of squash-merging, but perhaps that ship has sailed. Somewhat out of scope for discussion here anyway.

**Comment by zalberico** [comment]
The main bit I care about is `Fixes: ####` so people remember to put the issue link in there.

**Comment by barter-simsum** [comment]
@Fang-  I agree. I find pull request templates in general pretty annoying. Often pull request descriptions don't even really line up with the template given. Maybe just including a `Description` header -- does that even add anything? -- and I think the "Resolves#" string is fine to include and serves as a good reminder to link an issue. Will delete all comments etc.

How about that?

**Comment by barter-simsum** [comment]
The template is now a single line "Resovled #" as a basic reminder to link an issue. Doesn't require any annoying deletion of text from contributor.

@Fang-

**Comment by zalberico** [comment]
Cool - if we find people leaving the details blank we can ask them to add details (and if necessary introduce some templating if it's an issue).

**Comment by belisarius222** [comment]
Should this PR be targeting master, as it is now, or develop?

**Comment by mcevoypeter** [comment]
It should target `develop`. Good catch, @belisarius222.

**Review by mcevoypeter — APPROVED** [review]

**Comment by mcevoypeter** [comment]
@Fang- if you feel good about this, we can merge it.

**Comment by Fang-** [comment]
Sure, this is minimal enough. Looks better!

**Pull request merged**